rust/kernel/error.rs

```diff
@@ -76,7 +81,11 @@ impl Error {
 
         // INVARIANT: the check above ensures the type invariant
         // will hold.
-        Error(errno)
+        //
+        // Safety: `errno` must be non zero, which is guaranteed by the check
+        // above.
+        let nz_errno = unsafe { NonZeroI32::new_unchecked(errno) };
```

Comment:
Would it be better to call `from_kernel_errno_unchecked()` here to prevent having to write
```rust
Error(unsafe { NonZeroI32::new_unchecked(errno) })
```
twice?

Reply:
you mean
```rust
unsafe{ Error::from_kernel_errno_unchecked() }
```
?
it's not terser